crates/transport_tx5/src/test.rs

```diff
@@ -525,9 +536,161 @@ async fn preflight_send_recv() {
+        )
+        .await;
+
+    // We expect it to have timed out because no peer should be reachable at above URL
+    assert!(res.is_err());
+
+    // We expect the mark_peer_unresponsive() method on the TxSpaceHandler trait to have
+    // been invoked
+    let r = tokio::time::timeout(std::time::Duration::from_secs(5), async {
+        unresp_recv.recv().await
+    })
+    .await
+    .unwrap();
+
+    assert!(Some(faulty_url) == r);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn offline_peer_marked_unresponsive() {
+    // set the tx5 timeout to 5 seconds to keep the test reasonably short
+    let test = Test::new(None, Some(5)).await;
+
+    let (unresp_send, mut unresp_recv) = tokio::sync::mpsc::unbounded_channel();
+
+    let u1 = Arc::new(Mutex::new(Url::from_str("ws://bla.bla:38/1").unwrap()));
+    let u1_2 = u1.clone();
+    let h1 = Arc::new(CbHandler {
+        new_addr: Arc::new(move |url| {
+            *u1_2.lock().unwrap() = url;
+        }),
+        mark_peer_unresponsive: Arc::new({
+            move |url| {
+                unresp_send.send(url).map_err(|_| K2Error::Other {
+                    ctx: "Failed to send url to oneshot channel".into(),
+                    src: DynInnerError::default(),
+                })
+            }
+        }),
+        ..Default::default()
+    });
+    let t1 = test.build_transport(h1.clone()).await;
+    t1.register_space_handler(TEST_SPACE_ID, h1.clone());
+    t1.register_module_handler(TEST_SPACE_ID, "mod".into(), h1.clone());
+
+    let (s_send, mut s_recv) = tokio::sync::mpsc::unbounded_channel();
+    let u2 = Arc::new(Mutex::new(Url::from_str("ws://bla.bla:38/1").unwrap()));
+    let u2_2 = u2.clone();
+    let h2 = Arc::new(CbHandler {
+        new_addr: Arc::new(move |url| {
+            *u2_2.lock().unwrap() = url;
+        }),
+        recv_space_notify: Arc::new(move |url, space, data| {
+            let _ = s_send.send((url, space, data));
+            Ok(())
+        }),
+        ..Default::default()
+    });
+    let t2 = test.build_transport(h2.clone()).await;
+    t2.register_space_handler(TEST_SPACE_ID, h2.clone());
+    t2.register_module_handler(TEST_SPACE_ID, "mod".into(), h2.clone());
+
+    let u2: Url = u2.lock().unwrap().clone();
+    println!("got u2: {}", u2);
+
+    // check that send works initially while peer 2 is still online
+    t1.send_space_notify(
+        u2.clone(),
+        TEST_SPACE_ID,
+        bytes::Bytes::from_static(b"hello"),
+    )
+    .await
+    .unwrap();
+
+    // checks that recv works
+    let r = tokio::time::timeout(std::time::Duration::from_secs(5), async {
+        s_recv.recv().await
+    })
+    .await
+    .unwrap()
+    .unwrap();
+
+    println!("{r:?}");
+    assert_eq!(b"hello", r.2.as_ref());
+
+    // Check that the peer has not been marked as unresponsive
+    let r = tokio::time::timeout(std::time::Duration::from_secs(3), async {
+        unresp_recv.recv().await
```

Comment:
Consider using `unresp_recv.try_recv` here? I think a quick point in time check is acceptable because we got a response?

Reply:
Hmm, I'm not sure how to explain that. Perhaps the connection was idling even before the drop happened so the reduced wait before the drop means the same total amount of time is needed for the connection to get dropped. I'm not sure!